gravatar-quickeditor/src/main/java/com/gravatar/quickeditor/ui/avatarpicker/AvatarPicker.kt

```diff
@@ -146,10 +147,13 @@ internal fun AvatarPicker(uiState: AvatarPickerUiState, onEvent: (AvatarPickerEv
                     .fillMaxWidth()
                     .padding(bottom = 10.dp),
             )
-            ProfileCard(
-                profile = uiState.profile,
-                modifier = Modifier.padding(horizontal = 16.dp),
-            )
+            key(uiState.emailAvatars?.selectedAvatarId) {
```

Comment:
This one is new for me. What exactly is its usage here? Is it to only recompose when `selectedAvatarId` is changed? 

Won't that prevent updates when the `uiState.profile` is modified?

Reply:
> Is it to only recompose when selectedAvatarId is changed?

Yes. We know when the avatar is updated, but the Composable, which shows the avatar, doesn't depend on that state (the profile will be the same). So, using the key will force its recomposition based on the `selectedAvatarId` changes.

> Won't that prevent updates when the uiState.profile is modified

As far as I know, `ProfileCard` should still be updated if `uiState.profile` is modified.